packages/app-builder-lib/src/asar/asarUtil.ts

```diff
@@ -45,237 +43,115 @@ export class AsarPackager {
       fileSets[0],
     ].map(orderFileSet)
 
-    for (const fileSet of orderedFileSets) {
-      unpackedFileIndexMap.set(fileSet, await this.createPackageFromFiles(fileSet, packager.info))
-    }
-    await this.writeAsarFile(orderedFileSets, unpackedFileIndexMap)
-  }
-
-  private async createPackageFromFiles(fileSet: ResolvedFileSet, packager: Packager) {
-    const metadata = fileSet.metadata
-    // search auto unpacked dir
-    const unpackedDirs = new Set<string>()
-    const rootForAppFilesWithoutAsar = path.join(this.destination, "app")
+    const { unpackedPaths, copiedFiles } = await this.detectAndCopy(orderedFileSets, cancellationToken)
+    const unpackGlob = unpackedPaths.length > 1 ? `{${unpackedPaths.join(",")}}` : unpackedPaths.pop()
 
-    if (this.options.smartUnpack !== false) {
-      await detectUnpackedDirs(fileSet, unpackedDirs, this.unpackedDest, rootForAppFilesWithoutAsar)
+    let ordering = this.config.options.ordering || undefined
+    if (!ordering) {
+      // `copiedFiles` are already ordered due to `orderedFileSets` input, so we just map to their relative paths (via substring) within the asar.
+      const filesSorted = copiedFiles.map(file => file.substring(this.rootForAppFilesWithoutAsar.length))
+      ordering = await this.tmpDir.getTempFile({ prefix: "asar-ordering", suffix: ".txt" })
+      await fs.writeFile(ordering, filesSorted.join("\n"))
     }
 
-    const dirToCreateForUnpackedFiles = new Set<string>(unpackedDirs)
-
-    const correctDirNodeUnpackedFlag = async (filePathInArchive: string, dirNode: Node) => {
-      for (const dir of unpackedDirs) {
-        if (filePathInArchive.length > dir.length + 2 && filePathInArchive[dir.length] === path.sep && filePathInArchive.startsWith(dir)) {
-          dirNode.unpacked = true
-          unpackedDirs.add(filePathInArchive)
-          // not all dirs marked as unpacked after first iteration - because node module dir can be marked as unpacked after processing node module dir content
-          // e.g. node-notifier/example/advanced.js processed, but only on process vendor/terminal-notifier.app module will be marked as unpacked
-          await mkdir(path.join(this.unpackedDest, filePathInArchive), { recursive: true })
-          break
-        }
+    const options: CreateOptions = {
+      unpack: unpackGlob,
+      unpackDir: unpackGlob,
+      ordering,
+      dot: true,
+    }
+    // override logger temporarily to clean up console (electron/asar does some internal logging that blogs up the default electron-builder logs)
+    const consoleLogger = console.log
+    console.log = (...args) => {
+      if (args[0] === "Ordering file has 100% coverage.") {
```

Comment:
If the output content of the upstream electron asar log changes, we need to modify our side promptly as well.

Reply:
Very true. It's just a minor logs cleanup, so it's not breaking anything if they change the string. If they do, I'll eventually notice it and have to update it, but it's minor.

I could have just done `console.log = (...args) => log.info( { args }, "logging electron/asar")`, but felt the logs didn't need the extra entries